native/src/main/java/io/ballerina/stdlib/http/api/HttpDispatcher.java

```diff
@@ -424,6 +444,52 @@ static BObject createRequestContext(HttpCarbonMessage httpCarbonMessage) {
         return requestContext;
     }
 
+    private static void addJwtValuesToRequestContext(Runtime runtime, BObject requestContext, String authHeader) {
+        Object decodedJwt = invokeJwtDecode(runtime, authHeader);
+        if (Objects.nonNull(decodedJwt)) {
+            BMap requestCtxMembers = requestContext.getMapValue(REQUEST_CTX_MEMBERS);
+            requestCtxMembers.put(JWT_HEADER, ((BArray) decodedJwt).get(0));
+            requestCtxMembers.put(JWT_PAYLOAD, ((BArray) decodedJwt).get(1));
+        }
+    }
+
+    private static Object invokeJwtDecode(Runtime runtime, String authHeader) {
+        final Object[] jwtValues = new Object[1];
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        Callback decodeCallback = new Callback() {
```

Comment:
Are we silently skipping the error(JWT decode Error) returned from the invoke call? 
Also is it possible to add a negative test case where an error is returned from this async call?

Reply:
Yes. Since there is a posibility of `Bearer ....` header to be both `jwt` or `oauth2`, thought of ignoring the error.